client/client.go

```diff
@@ -35,6 +37,29 @@ var (
 	errorAddingServerCertificateToPool = errors.New("Error adding trusted server certificate to pool.")
 )
 
+var (
+	// 	                  http.Client.Timeout
+	// +--------------------------------------------------------+
+	// +--------+  +---------+  +-------+  +--------+  +--------+
+	// |  Dial  |  |   TLS   |  |Request|  |Response|  |Response|
+	// |        |  |handshake|  |       |  |headers |  |body    |
+	// +--------+  +---------+  +-------+  +--------+  +--------+
+	// +--------+  +---------+             +--------+
+	//  Dial        TLS                     Response
+	//  timeout     handshake               header
+	//  timeout                             timeout
+	//
+	//  It covers the entire exchange, from Dial (if a connection is not reused)
+	// to reading the body. This is to timeout long lasing connections.
+	//
+	// 4 hours shold be enough to download 2GB image file with the
+	// average download spead ~1 mbps
+	defaultClientReadingTimeout = 4 * time.Hour
+
+	// connection keepalive options
+	connectionKeepaliveTime = 10 * time.Second
```

Comment:
So 10 seconds is not the total keepalive time. 10s after the last packet was transmitted (`tcp_keepalive_time`) we will start sending keepalive packets. Those will be sent every 10s (`tcp_keepalive_intvl`) until reaching `tcp_keepalive_probes`. The last parameter is defined in `/proc/sys/net/ipv4/tcp_keepalive_probes` and by default this is 9. 
So having default settings this will be 100s which is even longer than you suggested.